torchmdnet/models/tensornet.py

```diff
@@ -163,17 +178,28 @@ def reset_parameters(self):
         self.out_norm.reset_parameters()
 
     def forward(
-        self, z, pos, batch, q: Optional[Tensor] = None, s: Optional[Tensor] = None
-    ):
+        self,
+        z: Tensor,
+        pos: Tensor,
+        batch: Tensor,
+        q: Optional[Tensor] = None,
+        s: Optional[Tensor] = None,
+    ) -> Tuple[Tensor, Optional[Tensor], Tensor, Tensor, Tensor]:
+
         # Obtain graph, with distances and relative position vectors
         edge_index, edge_weight, edge_vec = self.distance(pos, batch)
+        # This assert convinces TorchScript that edge_vec is a Tensor and not an Optional[Tensor]
+        assert (
+            edge_vec is not None
+        ), "Distance module did not return directional information"
+
         # Expand distances with radial basis functions
         edge_attr = self.distance_expansion(edge_weight)
         # Embedding from edge-wise tensors to node-wise tensors
         X = self.tensor_embedding(z, edge_index, edge_weight, edge_vec, edge_attr)
         # Interaction layers
-        for i in range(self.num_layers):
-            X = self.layers[i](X, edge_index, edge_weight, edge_attr)
+        for _, layer in enumerate(self.layers):
```

Comment:
`enumerate` is not needed

Reply:
Yes, otherwise its not TorchScript compatible

Reply:
I now get what you meant: https://pytorch.org/docs/stable/jit_language_reference.html#for-loops-over-constant-nn-modulelist